### PR TITLE
fix(replace): assign default size when replacing with expanded element

### LINKED ISSUE #1687: Transforming a Call Activity into a SubProcess does not resize the Shape
__Describe the Bug__

There is a certain edit sequence which is a little confusing because it leads to what seems like a dead end. 

__Steps to Reproduce__

1. Create Camunda 8 BPMN diagram
2. Create a task
3. Use wrench symbol to turn the task into a call activity
4. Now observe, that the wrench menu has additional entries (which is already a bit confusing): sub-process collapsed, sub-process expanded
5. Use wrench symbol to turn the call activity into an expanded sub process
6. Now you end up with an element that visually looks like a task. But when opening the wrench menu, it only allows me to change it into an event subprocess. So I cannot go back to turn it into a call activity, and I reached a dead end.

image

**Hints:** 
* I was unable to reproduce this exactly on [demo.bpmn.io](https://demo.bpmn.io/new)
* Was able to reproduce it in current web modeler 
* Was able to reproduce it in desktop modeler 5.0.0

__Expected Behavior__

* after step 3, I don't expect additional entries to appear in the wrench menu. Especially since it seems that the collapsed sub process is not fully supported UI wise in either desktop or web modeler. 
* When I use the wrench symbol to change the type of an element, I expect that I can use the same wrench symbol to change the type back

__Environment__

 - Browser: Firefox
 - OS: Ubuntu 21.04
 - Library version: Whatever is used by current web modeler and desktop modeler 5.0.0

### PR DESCRIPTION
**Pull request by philippfromme**
brave_iY6uJmCn6m

---

Closes #1687

### CODE REVIEW

File: lib/features/replace/BpmnReplace.js [hunk 1]
```diff
@@ -199,6 +199,18 @@ export default function BpmnReplace(
       // else if property is explicitly set, use it
       else if (target && has(target, 'isExpanded')) {
         newElement.isExpanded = target.isExpanded;
+
+        // assign default size of new expanded element
+        const defaultSize = elementFactory.getDefaultSize(newBusinessObject, {
```
Comment (marstamm) [inline lib/features/replace/BpmnReplace.js:204]: No const with PhantomJS ;)
```suggestion
        var defaultSize = elementFactory.getDefaultSize(newBusinessObject, {
```

**Comment by marstamm** [comment]
This works surprisingly well with tight Processes. A bit of manual layouting is required, but that's expected
Recording 2022-08-19 at 09 36 55

File: lib/features/replace/BpmnReplace.js (hunk 1 above)
Comment (philippfromme) [reply lib/features/replace/BpmnReplace.js:204]: Man, I've been working with our ES6 projects for too long. 🤡

**Comment by philippfromme** [comment]
> This works surprisingly well with tight Processes. A bit of manual layouting is required, but that's expected

https://github.com/philippfromme/diagram-js-element-detection to the rescue. 🚒 💨

File: lib/features/replace/BpmnReplace.js (hunk 1 above)
Comment (philippfromme) [reply lib/features/replace/BpmnReplace.js]: Fixed.

**Comment by marstamm** [comment]
> https://github.com/philippfromme/diagram-js-element-detection
> Do NOT use this in production!

This sign can't stop me because i can't read!

**Review by marstamm — APPROVED** [review]

**Comment by philippfromme** [comment]
> This sign can't stop me because i can't read!

Only use this for pet projects like the Camunda Modeler. 🐶

**Pull request merged**